Spatial transcriptomics techniques are revolutionizing our ability to understand tumour microenvironments by enabling the analysis of gene expression within its spatial context. Unlike conventional sequencing methods which require complete tissue dissociation, ST enables the examination of gene expression in situ, which preserves the spatial relationships among cells Current graph-based spatial transcriptomics methods, especially for domain identification, often suffer from oversmoothing, where deeper network layers produce increasingly homogeneous node representations and reduce model expressiveness. To overcome this challenge, we introduce ContraST, a self-supervised contrastive learning-based computational framework that encodes structural information of graphs into transformers to maintain distinct and informative node representations. Our model integrates a graph corruption strategy which randomizes gene expression representations across nodes while preserving the original graph structure. By modifying the readout function of Deep Graph Infomax with a global special node, it captures the graphâ€™s spatial relationships and gene expression patterns, enhancing clustering accuracy and spatial resolution in transcriptomics analysis. ContraST has been rigorously evaluated on a range of human and mouse tissue datasets, including Stereo-seq and Slide-seq samples of the brain, pancreatic cancer, and lymph node tissues. Our method achieves groundbreaking performance on three pivotal tasks, including spatial transcriptomics deconvolution at cell-type resolution, spatial domain identification, and multisample integration. We show that ContraST improves the precision and interpretability of spatial transcriptomics and sets a new benchmark for downstream tissue analysis.
